src/www/ui/api/Helper/UploadHelper.php

```diff
@@ -627,6 +628,31 @@ public function getUploadCopyrightList($uploadId)
     return $responseList;
   }
 
+  /**
+    *  Get the clearing status for files within an upload
+     * @param DbManager $dbManager DbManager object
+     * @param integer $uploadTreeTableId upload tree id of the file
+     * @param integer $groupId groupId of the user
+     * @return String String containing the Clearing status message
+*/
+
+  public function fetchClearingStatus($dbManager, $uploadTreeTableId, $groupId)
+  {
+    $clearingStatus = [];
+    $clearingStatusMsg = array("WIP",null,null,"TO_BE_DISCUSSED","IRRELEVANT","IDENTIFIED","DO_NOT_USE","NON_FUNCTIONAL");
+    $sql = "SELECT * FROM clearing_decision WHERE uploadtree_fk=$1 AND group_fk=$2 ORDER BY date_added DESC;";
+    $stmt = __METHOD__.'.collectMainLicenses';
```

Comment:
Does [`Fossology\Lib\Dao\ClearingDao::getFileClearings()`](https://github.com/fossology/fossology/blob/43924f05229e03bfa1c86eb285127f9f0b93d409/src/lib/php/Dao/ClearingDao.php#L159) not providing correct results?

Reply:
How do I get the `itemTreeBounds` object and pass it here?

Reply:
How do I get the `itemTreeBounds` object and pass it here?

Reply:
Using [Fossology\Lib\Dao\UploadDao::getItemTreeBounds()](https://github.com/fossology/fossology/blob/43924f05229e03bfa1c86eb285127f9f0b93d409/src/lib/php/Dao/UploadDao.php#L132)

And you can cache `uploadtreetablename` using [Fossology\Lib\Dao\UploadDao::getUploadtreeTableName()](https://github.com/fossology/fossology/blob/43924f05229e03bfa1c86eb285127f9f0b93d409/src/lib/php/Dao/UploadDao.php#L270) and pass it to `fetchClearingStatus()` (which is called inside a loop).

Reply:
PTAL @soham4abc

Reply:
Yes sir, I got this, discussed this with @shaheemazmalmmd. If you can provide me some more time we can implement both clearing status and the number of cleared files compared to total files response in the same PR itself. I have started working on it. If you allow me I will update this PR accordingly. Thank you for your review!! @GMishx

Reply:
Resolved